app/scripts/metamask-controller.js

```diff
@@ -2275,6 +2284,9 @@ export default class MetamaskController extends EventEmitter {
     this.txController.stopIncomingTransactionPolling();
     this.tokenDetectionController.disable();
 
+    // stop nftdetection controller from polling
+    this.nftDetectionController.stop();
```

Comment:
can you add a unit test for stopNetworkRequests function and expedct the `this.nftDetectionController.stop()` to be called ?